codegen-test/model/naming-obstacle-course-structs.smithy

```diff
@@ -0,0 +1,53 @@
+$version: "1.0"
+namespace naming_obs_structs
```

Comment:
I wanted to put this test into the existing `naming-obstacle-course.smithy`, but that one already defines `operation Result`, which would conflict unless it was in a different namespace.

I'm open to ideas here since I dislike the names here (ops vs structs) as they don't really make that much sense. Can we have a larger multi-namespace, multi-file, `naming-obstacle-course` service that still checks things such as the namespace being `crate`?

Reply:
not really, one service is one namespace. We could use renames but that would of course, defeat the purpose

Reply:
I think you can have nested namespaces in Smithy, so there's a chance this might work. No idea how our codegen handles it though.